bin/check_everything.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+"""
+This script verifies that the Submitty installation and course data
+directories are configured and permissioned correctly.
+
+This script is a work-in-progress.  Additional checks will be added or
+revised as needed.
+"""
+
+import os
+import pwd
+import stat
+import grp
+import sys
+
+
+# Will be filled in by INSTALL_SUBMITTY.sh
+SUBMITTY_INSTALL_DIR="__INSTALL__FILLIN__SUBMITTY_INSTALL_DIR__"
+SUBMITTY_DATA_DIR="__INSTALL__FILLIN__SUBMITTY_DATA_DIR__"
+
+HWPHP_USER="__INSTALL__FILLIN__HWPHP_USER__"
+HWCRON_USER="__INSTALL__FILLIN__HWCRON_USER__"
+
+COURSE_BUILDERS_GROUP="__INSTALL__FILLIN__COURSE_BUILDERS_GROUP__"
+
+###########################################################################
+def CheckItemBits (my_path, is_dir, my_owner, my_group, my_bits):
+    ret_val = True
+    try:
+        pwd.getpwnam(my_owner)
+    except KeyError:
+        sys.stderr.write("ERROR! user "+my_owner+" does not exist\n")
```

Comment:
`print("ERROR! user "+my_owner+" does not exist\n", file=sys.stderr)` is probably the more "pythonic" way to do this.